### PR TITLE
Add new-gclogentry and new-gclogmonitoredresource cmdlets

### PR DESCRIPTION
**Pull request by quoctruong**
This change is [Reviewable](https://reviewable.io/reviews/googlecloudplatform/google-cloud-powershell/385)

### CODE REVIEW

**Comment by quoctruong** [comment]
@iantalarico @ILMTitan PTAL

**Comment by iantalarico** [comment]
Unless @ILMTitan is short on time I'll let him take this.  He has a lot more experience and knowledge with Powershell then I do.

File: Google.PowerShell/Logging/GcLogCmdlet.cs [hunk 1]
```diff
@@ -359,4 +359,277 @@ protected override void ProcessRecord()
...
+        /// <summary>
+        /// This dynamic parameter dictionary is used by PowerShell to generate parameters dynamically.
+        /// </summary>
+        private RuntimeDefinedParameterDictionary dynamicParameters;
+
+        /// <summary>
+        /// This function is part of the IDynamicParameters interface.
+        /// PowerShell uses it to generate parameters dynamically.
+        /// </summary>
+        public object GetDynamicParameters()
```
Comment (ILMTitan) [inline Google.PowerShell/Logging/GcLogCmdlet.cs]: If this parameter always exists, why is this a dynamic parameter? A comment explaining would be useful.

**Comment by quoctruong** [comment]
@ILMTitan PTAL

**Comment by ILMTitan** [comment]
LGTM

**Comment by quoctruong** [comment]
Thanks!

**Pull request merged**